core-java/src/test/java/org/baeldung/java/collections/JoinSplitCollectionsUnitTest.java

```diff
@@ -0,0 +1,178 @@
+	}
+	
+	@Test
+	public void whenJoiningTwoCollections_thenJoined() {
+		Collection<Integer> collection1 = Arrays.asList(7, 8, 9);
+		Collection<Integer> collection2 = Arrays.asList(10, 11, 12);
+		Collection<Integer> result = Stream.concat(collection1.stream(), collection2.stream())
+				.collect(Collectors.toList());
+	
+		assertTrue(result.equals(Arrays.asList(7, 8, 9, 10, 11, 12)));
+	}
+
+	@Test
+	public void whenJoiningTwoCollectionsWithFilter_thenJoined() {
+		Collection<Integer> collection1 = Arrays.asList(7, 8, 11);
+		Collection<Integer> collection2 = Arrays.asList(9, 12, 10);
+		Collection<Integer> result = Stream.concat(collection1.stream(), collection2.stream())
+				.filter(next -> next <= 10)
+				.collect(Collectors.toList());
+	
+		assertTrue(result.equals(Arrays.asList(7, 8, 9, 10)));
+	}
+
+	@Test
+	public void whenConvertArrayToString_thenConverted() {
+		String[] colors = new String[] { "Red", "Blue", "Green", "Yellow" };
+		String result = Arrays.stream(colors)
+				.collect(Collectors.joining(", "));
+
+		assertEquals(result, "Red, Blue, Green, Yellow");
+	}
+
+	@Test
+	public void whenConvertCollectionToString_thenConverted() {
+		Collection<String> directions = Arrays.asList("Left", "Right", "Top", "Bottom");
+		String result = directions.stream()
+				.collect(Collectors.joining(", "));
+	
+		assertEquals(result, "Left, Right, Top, Bottom");
+	}
+
+	@Test
+	public void whenConvertMapToString_thenConverted() {
+		Map<Integer, String> users = new HashMap<>();
+		users.put(1, "John Doe");
+		users.put(2, "Paul Smith");
+		users.put(3, "Susan Anderson");
+
+		String result = users.entrySet().stream().map(entry -> entry.getKey() + " = " + entry.getValue())
+				.collect(Collectors.joining(", "));
+
+		assertEquals(result, "1 = John Doe, 2 = Paul Smith, 3 = Susan Anderson");
+	}
+
+	@Test
+	public void whenConvertNestedCollectionToString_thenConverted() {
+		Collection<List<String>> nested = new ArrayList<>();
+		nested.add(Arrays.asList("Left", "Right", "Top", "Bottom"));
+		nested.add(Arrays.asList("Red", "Blue", "Green", "Yellow"));
+	
+		String result = nested.stream()
+				.map(nextList -> nextList.stream()
+						.collect(Collectors.joining("-")))
+				.collect(Collectors.joining("; "));
+	
+		assertEquals(result, "Left-Right-Top-Bottom; Red-Blue-Green-Yellow");
+	}
+
+	@Test
+	public void whenConvertCollectionToStringAndSkipNull_thenConverted() {
+		Collection<String> fruits = Arrays.asList("Apple", "Orange", null, "Grape");
+		String result = fruits.stream()
+				.filter(next -> next != null)
+				.collect(Collectors.joining(", "));
+	
+		assertEquals(result, "Apple, Orange, Grape");
+	}
+
+	@Test
+	public void whenSplitCollectionHalf_thenConverted() {
+		Collection<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+		Collection<Integer> result1 = new ArrayList<>();
+		Collection<Integer> result2 = new ArrayList<>();
+		AtomicInteger count = new AtomicInteger();
```

Comment:
Since the Streams are not concurrent it doesn't really need to be an AtomicInteger.
It becomes indispensable once the stream is parallel. So I guess it can be changed here to int, but it is a good practice and future-proof to use an AtomicInteger.